client/dashboard/index.js

```diff
@@ -2,93 +2,54 @@
- */
-import { getSetting } from '@woocommerce/wc-admin-settings';
-import { getNewPath, updateQueryString } from '@woocommerce/navigation';
-
 /**
  * Internal dependencies
  */
 import './style.scss';
 import CustomizableDashboard from './customizable';
 import ProfileWizard from './profile-wizard';
+import CartModal from './profile-wizard/cart-modal';
 import withSelect from 'wc-api/with-select';
-import { isOnboardingEnabled } from 'dashboard/utils';
+import { isOnboardingEnabled, getProductIdsForCart } from 'dashboard/utils';
 
 class Dashboard extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
+			showCartModal: false,
+		};
+	}
 	componentDidUpdate( prevProps ) {
 		const profileItems = get( this.props, 'profileItems', {} );
 		const prevProfileItems = get( prevProps, 'profileItems', {} );
 
+		// We only want to show the modal during the transition between the profile wizard and dashboard and if products were selected.
 		if ( profileItems.completed && ! prevProfileItems.completed ) {
-			updateQueryString( {}, '/', {} );
-			this.redirectToCart();
-		}
-	}
-
-	getProductIds() {
-		const productIds = [];
-		const profileItems = get( this.props, 'profileItems', {} );
-		const onboarding = getSetting( 'onboarding', {} );
-		const productTypes = profileItems.product_types || [];
-
-		productTypes.forEach( productType => {
-			if (
-				onboarding.productTypes[ productType ] &&
-				onboarding.productTypes[ productType ].product
-			) {
-				productIds.push( onboarding.productTypes[ productType ].product );
+			const productIds = getProductIdsForCart( profileItems );
+			if ( productIds.length ) {
+				/* eslint-disable react/no-did-update-set-state */
+				this.setState( { showCartModal: true } );
+				/* eslint-enable react/nod-ddi-update-set-state */
 			}
-		} );
-
-		const theme = onboarding.themes.find( themeData => themeData.slug === profileItems.theme );
-
-		if ( theme && theme.id && ! theme.is_installed ) {
-			productIds.push( theme.id );
 		}
-
-		return productIds;
-	}
-
-	redirectToCart() {
-		if ( ! isOnboardingEnabled() ) {
-			return;
-		}
-
-		const productIds = this.getProductIds();
-		const backPath = getNewPath( {}, '/', {} );
-		const { connectNonce } = getSetting( 'onboarding', {} );
-
-		if ( ! productIds.length ) {
-			return;
-		}
-
-		document.body.classList.add( 'woocommerce-admin-is-loading' );
-
-		const url = addQueryArgs( 'https://woocommerce.com/cart', {
-			'wccom-site': getSetting( 'siteUrl' ),
-			'wccom-woo-version': getSetting( 'wcVersion' ),
-			'wccom-replace-with': productIds.join( ',' ),
-			'wccom-connect-nonce': connectNonce,
-			'wccom-back': backPath,
-		} );
-		window.location = url;
 	}
 
 	render() {
 		const { path, profileItems, query } = this.props;
+		const { showCartModal } = this.state;
 
 		if ( isOnboardingEnabled() && ! profileItems.completed ) {
 			return <ProfileWizard query={ query } />;
 		}
 
+		if ( isOnboardingEnabled() && showCartModal ) {
```

Comment:
The only issue with separating the modal from the profile wizard here is the theme previews can't be shown in the background.  May not be super important though; /cc @jameskoster in case you have a strong opinion one way or the other.

#### Current
<img width="997" alt="Screen Shot 2019-12-19 at 2 17 54 PM" src="https://user-images.githubusercontent.com/10561050/71150019-338dd680-226b-11ea-8721-d8ff9bbeae41.png">


#### Design
<img width="647" alt="Screen Shot 2019-12-19 at 2 23 45 PM" src="https://user-images.githubusercontent.com/10561050/71150023-3688c700-226b-11ea-8ab8-8d60ebead744.png">

Reply:
I'm moving some things around to make this possible.

Reply:
I have to move some things around to make this work (namely mark the wizard as completed at a different point) but this now opens in front of the theme step.

https://github.com/woocommerce/woocommerce-admin/pull/3444/commits/d7762abe8ad6717ef3db01792f1b1fe9f3584906